// @vcsuite/eslint-config/common.js
module.exports = {
  plugins: [
    'import',
  ],
  rules: {
    indent: [
      'error',
      2,
      {
        ignoredNodes: [
          'TemplateLiteral',
        ],
        SwitchCase: 1,
      },
    ],
    'template-curly-spacing': [
      'off',
    ],
    'arrow-body-style': 'off',
    'no-bitwise': 'off',
    'no-console': 2,
    'import/extensions': [
      'warn',
      'ignorePackages',
    ],
    'import/no-cycle': [
      'error',
      {
        ignoreExternal: true,
      },
    ],
    'max-len': [
      'error',
      120,
      2,
      {
        ignoreUrls: true,
        ignoreComments: true,
        ignoreRegExpLiterals: true,
        ignoreStrings: true,
        ignoreTemplateLiterals: true,
      },
    ],
    'linebreak-style': 0,
    'no-underscore-dangle': [
      'error',
      {
        allow: [
          '_id',
          '_batchId',
        ],
        allowAfterThis: true,
        allowAfterSuper: true,
      },
    ],
    'no-plusplus': [
      'error',
      {
        allowForLoopAfterthoughts: true,
      },
    ],
    'no-param-reassign': [
      'error',
      {
        props: false,
      },
    ],
    'prefer-destructuring': [
      'error',
      {
        VariableDeclarator: {
          array: false,
          object: true,
        },
        AssignmentExpression: {
          array: false,
          object: true,
        },
      },
      {
        enforceForRenamedProperties: false,
      },
    ],
    'operator-linebreak': [
      'error',
      'after',
    ],
    'no-multiple-empty-lines': [
      'error',
      {
        max: 2,
        maxEOF: 1,
      },
    ],
    'object-curly-newline': [
      'error',
      {
        ObjectExpression: {
          minProperties: 6,
          multiline: true,
          consistent: true,
        },
        ObjectPattern: {
          minProperties: 6,
          multiline: true,
          consistent: true,
        },
        ImportDeclaration: {
          minProperties: 6,
          multiline: true,
          consistent: true,
        },
        ExportDeclaration: {
          minProperties: 6,
          multiline: true,
          consistent: true,
        },
      },
    ],
    'no-else-return': 'off',
    'arrow-parens': [2, 'as-needed', { requireForBlockBody: true }],
  },
};
