// @vcsuite/eslint-config/jsdoc.js
module.exports = {
  plugins: [
    'jsdoc',
  ],
  extends: [
    'plugin:jsdoc/recommended',
  ],
  settings: {
    jsdoc: {
      tagNamePreference: {
        augments: 'extends',
        constant: 'const',
      },
      preferredTypes: {
        object: 'Object',
      },
    },
  },
  rules: {
    'jsdoc/check-access': 2,
    'jsdoc/check-alignment': 2,
    'jsdoc/check-param-names': 2,
    'jsdoc/check-property-names': 2,
    'jsdoc/check-tag-names': [2, { definedTags: ['api', 'export', 'memberOf', 'template'] }],
    'jsdoc/check-types': 2,
    'jsdoc/check-values': 2,
    'jsdoc/empty-tags': 2,
    'jsdoc/newline-after-description': 0,
    'jsdoc/no-undefined-types': 0,
    'jsdoc/require-jsdoc': 0, // XXX implement at some point
    'jsdoc/require-param': [2, { enableFixer: false }],
    'jsdoc/require-param-name': 2,
    'jsdoc/require-param-type': 2,
    'jsdoc/require-param-description': 0,
    'jsdoc/require-property': 2,
    'jsdoc/require-property-name': 2,
    'jsdoc/require-property-type': 2,
    'jsdoc/require-property-description': 0,
    'jsdoc/require-returns': 2,
    'jsdoc/require-returns-check': 2,
    'jsdoc/require-returns-description': 0,
    'jsdoc/require-returns-type': 2,
    'jsdoc/valid-types': 0,
  },
};
